### PR TITLE
feat: setup prisma and db migration

### PR DESCRIPTION
**Pull request by bharathraj25**
Completed Setting up Prisma & DB schema, along generated db migration files.

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model action_items {
+  item_id         Int       @unique @default(autoincrement())
```
Comment (sugumar-mani) [inline prisma/schema.prisma:14]: it can be note_id or id

File: prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  is_done    Boolean   @default(false)
```
Comment (sugumar-mani) [inline prisma/schema.prisma:15]: status with enum values 'COMPLETED' 'PENDING'

File: prisma/schema.prisma [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  content    String
```
Comment (sugumar-mani) [inline prisma/schema.prisma:16]: can be 'note

File: prisma/schema.prisma [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  jira_link  String?
+  created_at DateTime  @default(now())
+  due_date   DateTime?
+  isValid    Boolean   @default(true)
```
Comment (sugumar-mani) [inline prisma/schema.prisma:20]: can be 'deleted'

File: prisma/schema.prisma [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+model key_decisions {
```
Comment (sugumar-mani) [inline prisma/schema.prisma:23]: we don't need multiple tables. just have 'po_notes' table

File: prisma/schema.prisma [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model action_items {
```
Comment (sugumar-mani) [inline prisma/schema.prisma:13]: rename table to PO Notes

Comment (bharathraj25) [reply prisma/schema.prisma]: DONE

File: prisma/schema.prisma (hunk 5 above)
Comment (bharathraj25) [reply prisma/schema.prisma]: DONE

File: prisma/schema.prisma (hunk 4 above)
Comment (bharathraj25) [reply prisma/schema.prisma]: DONE

File: prisma/schema.prisma (hunk 3 above)
Comment (bharathraj25) [reply prisma/schema.prisma]: DONE

File: prisma/schema.prisma (hunk 2 above)
Comment (bharathraj25) [reply prisma/schema.prisma]: DONE

File: prisma/schema.prisma (hunk 1 above)
Comment (bharathraj25) [reply prisma/schema.prisma]: DONE

File: prisma/schema.prisma [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model PoNotes {
```
Comment (sugumar-mani) [inline prisma/schema.prisma]: can be `PONotes` notes instead of PoNotes

File: prisma/schema.prisma [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  noteId    Int       @unique @default(autoincrement())
+  type      Type
+  status    Status    @default(PENDING)
+  note      String
+  jiraLink  String?
```
Comment (sugumar-mani) [inline prisma/schema.prisma:18]: lets not bind to jira. can we name it to `issueLink` ?

File: prisma/migrations/20230128074340_init/migration.sql [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+-- DropTable
+DROP TABLE "key_decisions";
+
+-- CreateTable
+CREATE TABLE "PoNotes" (
+    "noteId" SERIAL NOT NULL,
+    "type" "Type" NOT NULL,
+    "status" "Status" NOT NULL DEFAULT 'PENDING',
+    "note" TEXT NOT NULL,
```
Comment (sugumar-mani) [inline prisma/migrations/20230128074340_init/migration.sql:29]: lets make 1000 chars limit for now

File: prisma/migrations/20230128074340_init/migration.sql [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    "jiraLink" TEXT,
```
Comment (sugumar-mani) [inline prisma/migrations/20230128074340_init/migration.sql:30]: 500 char limit for now

**Pull request merged**